cmd/mailbox-controller/controller.go

```diff
@@ -255,76 +258,82 @@ func (ctl *mbCtl) sync(ctx context.Context, refany any) bool {
-	if err != nil {
-		logger.Error(err, "Unable to start executing the script")
-		return true
-	}
-	outBuf := bufio.NewReader(stdout)
-	errBuf := bufio.NewReader(stderr)
+	resourcesToBind := []string{"syncerconfigs", "edgesyncconfigs"}
+	for idx, resource := range resourcesToBind {
+		logger.V(2).Info("Ensuring binding", "script", shellScriptName, "resource", resource)
 
-	r, _ := regexp.Compile("🔒 (Created|Updated) secret.*namespace (.*)")
-	found, namespaceName := false, ""
-	for {
-		line, _, err := errBuf.ReadLine()
+		// suffix helps isolate this controller's multiple workers to prevent concurrent access of a single kubeconfig file.
+		// This is a compromise due to the situation that 1) we are using kcp and 2) we are running tests on bare process of this controller.
+		suffix := atomic.AddUint64(&initialSuffix, 1)
+
+		makeCopyOfKubeConfig := fmt.Sprintf("cp $KUBECONFIG $KUBECONFIG.copy%d", suffix)
+		removeCopyWhenExits := fmt.Sprintf("trap 'rm $KUBECONFIG.copy%d' EXIT", suffix)
+		invokeScript := strings.Join([]string{
+			fmt.Sprintf("KUBECONFIG=$KUBECONFIG.copy%d", suffix),
+			shellScriptName,
+			workspace.Name,
+			resource,
+		}, " ")
+		if idx == 0 {
+			invokeScript = invokeScript + " --start-konnector true"
+		}
+		cmdLine := strings.Join([]string{
+			makeCopyOfKubeConfig,
+			removeCopyWhenExits,
+			invokeScript,
+		}, "; ")
+		logger.V(2).Info(cmdLine)
+		cmd := exec.Command("/bin/sh", "-c", cmdLine)
+		stdout, err := cmd.StdoutPipe()
 		if err != nil {
-			logger.V(4).Info("End of stderr")
-			break
+			logger.Error(err, "Unable to return a pipe for stdout")
+			return true
 		}
-		logger.V(2).Info(string(line))
-		// kube-bind put the information that we interesed in (the cluster namespace) in stderr
-		matches := r.FindStringSubmatch(string(line))
-		if len(matches) == r.NumSubexp()+1 {
-			found, namespaceName = true, matches[len(matches)-1]
+		stderr, err := cmd.StderrPipe()
+		if err != nil {
+			logger.Error(err, "Unable to return a pipe for stderr")
+			return true
 		}
-	}
-	if found {
-		logger.V(2).Info("Kube-bind cluster namespace", "namespaceName", namespaceName)
-	}
-
-	// There is a contact between the script and the code that this message is the signal for successful bindings
-	r, _ = regexp.Compile("CRDs already in place, returning")
-	established := false
-	for {
-		line, _, err := outBuf.ReadLine()
+		err = cmd.Start()
 		if err != nil {
-			logger.V(4).Info("End of stdout")
-			break
+			logger.Error(err, "Unable to start executing the script")
+			return true
 		}
-		logger.V(2).Info(string(line))
-		if r.MatchString(string(line)) {
-			established = true
+		outBuf := bufio.NewReader(stdout)
+		errBuf := bufio.NewReader(stderr)
+		for {
+			line, _, err := outBuf.ReadLine()
+			if err != nil {
+				if err == io.EOF {
+					logger.V(4).Info("End of stdout")
+					break
+				} else {
+					logger.Error(err, "Unable to read stdout")
+					return true
+				}
+			}
+			logger.V(2).Info(string(line))
```

Comment:
I would add an explanation of what this is. For example:
```
logger.V(2).InfoS("Stdout from kube-bind", "resource", resource, "line", line)
```

Reply:
Changed in latest push.